internal/broker/bind_get.go

```diff
@@ -22,7 +27,27 @@ func (b *GetBindingEndpoint) GetBinding(_ context.Context, instanceID, bindingID
 	b.log.Infof("GetBinding instanceID: %s", instanceID)
 	b.log.Infof("GetBinding bindingID: %s", bindingID)
 
+	binding, err := b.bindings.Get(instanceID, bindingID)
+
+	if binding == nil {
+		message := "Binding not found"
+		return domain.GetBindingSpec{}, apiresponses.NewFailureResponse(fmt.Errorf(message), http.StatusNotFound, message)
+	}
+
+	if binding.InstanceID != instanceID {
+		message := "Binding not found"
+		return domain.GetBindingSpec{}, apiresponses.NewFailureResponse(fmt.Errorf(message), http.StatusNotFound, message)
+	}
+
+	if err != nil {
+		b.log.Errorf("GetBinding error: %s", err)
+		message := "Error while getting binding"
+		return domain.GetBindingSpec{}, apiresponses.NewFailureResponse(fmt.Errorf(message), http.StatusInternalServerError, message)
```

Comment:
We do not pass any information from the `err` to the caller, we use it only in log message.